framework/src/packets/ip/v6/srh.rs

```diff
@@ -178,33 +151,16 @@ impl<E: Ipv6Packet> SegmentRouting<E> {
     }
 
     #[inline]
-    pub fn segments_left(&self) -> u8 {
-        self.header().segments_left
+    pub fn set_routing_type(&mut self, routing_type: u8) {
+        self.header_mut().routing_type = routing_type;
     }
 
-    /// Sets the segments left and the IPv6 `dst` to match.
-    ///
-    /// Given a segment list of 3 segments, `[a, b, c]`, if the
-    /// `segments_left` is set to 2, then the IPv6 `dst` is set
-    /// to the (n + 1)th segment in the list, which is `c`.
-    ///
-    /// # Remarks
-    ///
-    /// `segments` should be set before setting `segments_left`
-    /// because the new `dst` value is taken from the segments
-    /// list.
-    #[inline]
-    pub fn set_segments_left_and_dst(&mut self, segments_left: u8) -> Result<()> {
-        if let Some(&segment) = self.segments().get(segments_left as usize) {
-            self.header_mut().segments_left = segments_left;
-            self.envelope_mut().set_dst(IpAddr::V6(segment))
-        } else {
-            Err(SegmentsOutOfBoundsError(segments_left as usize, self.segments().len()).into())
-        }
+    #[inline]
+    pub fn segments_left(&self) -> u8 {
+        self.header().segments_left
     }
 
-    /// Sets only the segments left. To keep segments left and IPv6 dst in sync,
-    /// @see set_segments_left_and_dst.
+    #[inline]
     pub fn set_segments_left(&mut self, segments_left: u8) {
```

Comment:
@drunkirishcoder I may keep a comment in here about keeping things in sync w/ dst.